### DISCUSSION TITLE
Missing wxfluidpatcher.pyw

### DISCUSSION THREAD

**Discussion by BeFrankly** [GeekFunkLabs/fluidpatcher#49] [Q&A] [answered]
Hello,
first I want to thank you for such a great software, especially for my Raspi in junction with my Novation Launchkey 25 :-)

I found [this video you made](https://geekfunklabs.com/2020/08/15/fluidpatcher-desktop-mode/) and was not able to find the script wxfluidpatcher.pyw to remotely configure the raspberry pi. I just found fluidpatcher.py but it has no remote mode. So were has it gone or to say: How to remotely configure Raspi (not with ssh and vi).
Regards
Frank

**Comment by albedozero** [comment, marked as answer]
Glad to hear you're getting some use from the software! I renamed _wxfluidpatcher.pyw_ to _fluidpatcher.pyw_ some time ago, in case I decide at some point to rewrite it using a GUI toolkit other than wxpython, such as PyQt perhaps, but also to make it clear that this program is the definitive desktop version of fluidpatcher.

I also removed the network link features in 5adfbca. Controlling and editing patches from the network and the Pi at the same time could easily create conflicts/confusion if the user switched to a different patch/bank on the Pi when editing over the network. It also made the code more cumbersome and unclear, and I'm striving to make these scripts clear enough for a user to modify to their own purposes. I also had a [browser-based file manager](https://youtu.be/k6_rTFHzgo0?t=90) solution in place already for editing banks that is more robust and secure. If you don't have this installed, you can get it by running the install script (`curl -L git.io/squishbox | bash`) and replying "No/No Change" to the other options.

**Comment by BeFrankly** [comment]
Dear @albedozero ,
I know what you mean and it's a good idea to keep thinks clear. But, it's a shame that you removed that network link feature. The video has been an appetizer for me and I thought:" Yeah, great, that's what I want" but it ended up here telling me, that the feature is not there anymore. :-(
The browser-based manager is a good start but it misses the feature of investigating the sound files, test them and automatic reload of the bank file. Ok. That's how it is.

**Reply by albedozero** [reply]
On SquishBox when you go the Load Bank option, the first choice is the current bank, so you can quickly reload a bank that you modified using the browser editor with just a couple clicks. Maybe it would be good to add the option for a BANK_RELOAD button to headlesspi.py to get the same kind of behavior. If you only have a few banks, you can fake it now by hitting the BANK_INC button until you get back to the current bank, but I admit this isn't as convenient.

I do most of my patch creating in fluidpatcher.pyw on a Windows or Linux desktop (preferably Linux since the LADSPA effects work there), and then copy the files over using the browser file manager. For some reason, this always seemed faster and more comfortable than using the netlink. Maybe because I can just open up the program and start editing, and I don't have the additional step of making sure my Pi is plugged in and connected to an audio output. Now that I recall more, it was also not too difficult to crash the software on the Pi by sending an erroneous bank file, and the netlink would sometimes disconnect inexplicably. I want to do a newer video showing off the best way(s) to use fluidpatcher.pyw - then maybe I can unlist that older video with the netlink tease in it ;-)

I appreciate your feedback. Sorry for the long reply - this is mostly for my own documentation/memory :-P
